app/controllers/tree_display_controller.rb

```diff
@@ -5,124 +5,53 @@ def action_allowed?
-  def goto_author_feedbacks
-    node_object = TreeFolder.find_by_name('Author Feedback')
-    session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
-    redirect_to :controller => 'tree_display', :action => 'list'
-  end
-
-  # direct access to global survey
-  def goto_global_survey
-    node_object = TreeFolder.find_by_name('Global Survey')
-    session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
-    redirect_to :controller => 'tree_display', :action => 'list'
-  end
-
-  # direct access to surveys
-  def goto_surveys
-    node_object = TreeFolder.find_by_name('Survey')
-    session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
-    redirect_to :controller => 'tree_display', :action => 'list'
-  end
-
-  # direct access to course evaluations
-  def goto_course_evaluations
-    node_object = TreeFolder.find_by_name('Course Evaluation')
-    session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
-    redirect_to :controller => 'tree_display', :action => 'list'
-  end
-
-  # direct access to courses
-  def goto_courses
-    node_object = TreeFolder.find_by_name('Courses')
-    session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
-    redirect_to :controller => 'tree_display', :action => 'list'
-  end
-
   def goto_bookmarkrating_rubrics
     node_object = TreeFolder.find_by_name('Bookmarkrating')
     session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
     redirect_to :controller => 'tree_display', :action => 'list'
   end
 
-  # direct access to assignments
-  def goto_assignments
-    node_object = TreeFolder.find_by_name('Assignments')
-    session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
-    redirect_to :controller => 'tree_display', :action => 'list'
-  end
+  # direct access
+ def go_to_menu_items
+    name = params[:params1]
+    if name
+      if name == "Review rubrics"
+          name = "Review"
+        elsif name == "Teammate review rubrics"
+          name = "Teammate Review"
+        elsif name == "Metareview rubrics"
+          name = "Metareview"
+        elsif name == "Author feedbacks"
+          name = "Author Feedback"
+        elsif name == "Global surveys"
+          name = "Global Survey"
+        elsif name == "Course evaluations"
+          name = "Course Evaluation"
+        elsif name == "Surveys"
+            name = "Survey"
+      end
+
+      node_object = TreeFolder.find_by_name(name)
+      puts node_object.inspect
+      session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
+      if node_object.name == "Courses"
+        session[:last_open_tab] = 1
+      elsif node_object.name == "Assignments"
+        session[:last_open_tab] = 2
+      elsif node_object.id ==1 ||node_object.id ==4 ||node_object.id ==5 ||node_object.id ==6 ||node_object.id ==7 ||node_object.id ==8 ||node_object.id ==9 ||node_object.id ==10 # if node_object_name is Questionnaires or its child_nodes
```

Comment:
Are we allowed to make the changes now? Or you were just informing us about how to do it in a correct way?

Thanks,
Anush
 
> On Nov 23, 2015, at 1:02 PM, Zhewei Hu <notifications@github.com> wrote:
> 
> In app/controllers/tree_display_controller.rb <https://github.com/expertiza/expertiza/pull/592#discussion_r45636894>:
> 
> > +        elsif name == "Global surveys"
> > +          name = "Global Survey"
> > +        elsif name == "Course evaluations"
> > +          name = "Course Evaluation"
> > +        elsif name == "Surveys"
> > +            name = "Survey"
> > +      end
> > +
> > +      node_object = TreeFolder.find_by_name(name)
> > +      puts node_object.inspect
> > +      session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
> > +      if node_object.name == "Courses"
> > +        session[:last_open_tab] = 1
> > +      elsif node_object.name == "Assignments"
> > +        session[:last_open_tab] = 2
> > +      elsif node_object.id ==1 ||node_object.id ==4 ||node_object.id ==5 ||node_object.id ==6 ||node_object.id ==7 ||node_object.id ==8 ||node_object.id ==9 ||node_object.id ==10 # if node_object_name is Questionnaires or its child_nodes
> Hardwired record numbers, should be retrieved from DB.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/expertiza/expertiza/pull/592/files#r45636894>.
>